src/screens/signUp/index.js

```diff
@@ -1,67 +1,145 @@
 import React, {PureComponent} from 'react';
-import {View, Button} from '../../components';
+import {View, Button, Touchable, Text} from '../../components';
 import {Input} from 'react-native-elements';
 import {signUp} from '../../store/actions';
 import {connect} from 'react-redux';
 import NavigationService from '../../services/NavigationService';
-import {styles} from './styles'
+import {styles} from './styles';
+import * as values from '../../constants/values';
+import * as schema from '../../constants/validationSchema';
+import i18n from '../../localization';
+
 
 class SignUp extends PureComponent {
+
+  static navigationOptions = {
+    headerLeft: (
+      <Touchable
+        onPress={() => NavigationService.pop(1)}
+        style={styles.backButton}>
+        <Text style={styles.backText}>{i18n.t('BACK_BUTTON')}</Text>
+      </Touchable>
+    ),
+  };
+
   state = {
     username: '',
     email: '',
     password: '',
+    isValid: {usernameError: true, emailError: true, passwordError: true},
   };
-  
+
+  validation = (username, email, password) => {
+    let usernameError = schema.schemaUsername.isValidSync({
+      name: username,
+    });
+    let emailError = schema.schemaEmail.isValidSync({
+      email: email,
+    });
+    let passwordError = schema.schemaPassword.isValidSync({
+      password: password,
+    });
+    const isValid = {
+      usernameError,
+      emailError,
+      passwordError,
+    };
+    if (usernameError && emailError && passwordError) {
+      this.setState({isValid: isValid});
+      return true;
+    } else {
+      this.setState({isValid: isValid});
+      return false;
+    }
+  };
+
   handleClick = () => {
-    this.props.onSignUp(
-      this.state.username.trim(),
-      this.state.email.trim(),
-      this.state.password,
-    );
-    NavigationService.navigate('Auth', {});
+    const username = this.state.username.trim();
+    const email = this.state.email.trim();
+
+    const password = this.state.password;
+    const isValid =this.validation(username, email, password);
+    if (isValid) {
+      this.props.onSignUp(username, email, password);
+      NavigationService.navigate('Auth', {});
+    }
+  };
+
+  handleChangeEmail = () => email => {
+    this.setState({email});
+  };
+
+  handleChangeUsername = () => username => {
+    this.setState({username});
+  };
+
+  handleChangePassword = () => password => {
+    this.setState({password});
   };
 
   render() {
     return (
       <View style={styles.container}>
         <Input
           style={styles.input}
-          onChangeText={username => this.setState({username})}
+          onChangeText={this.handleChangeUsername()}
           value={this.state.username}
-          placeholder={'Username'}
+          placeholder={i18n.t('LOGIN.USERNAME')}
         />
+        {!this.state.isValid.usernameError && (
```

Comment:
Write separate component, that will received this value, handleChange and error.